dae/dae/pheno/pheno_import.py

```diff
@@ -0,0 +1,824 @@
+    assert not split_path[0], f"'{phenotype_data_name}' is a directory path!"
+    return phenotype_data_name
+
+
+def generate_phenotype_data_config(
+    args: argparse.Namespace, regressions: Any,
+) -> dict[str, Any]:
+    """Construct phenotype data configuration from command line arguments."""
+    dbfile = os.path.join("%(wd)s", os.path.basename(args.pheno_db_filename))
+    # pheno_db_path = os.path.dirname("%(wd)s")  # noqa
+    config = {
+        "vars": {"wd": "."},
+        "phenotype_data": {
+            "name": args.pheno_name,
+            "dbfile": dbfile,
+            "browser_images_url": "static/images/",
+        },
+    }
+    if regressions:
+        regressions_dict = regressions.to_dict()
+        for reg in regressions_dict["regression"].values():
+            if reg["measure_name"] is None:
+                del reg["measure_name"]
+            if reg["measure_names"] is None:
+                del reg["measure_names"]
+        config["regression"] = regressions_dict["regression"]
+    return config
+
+
+def parse_phenotype_data_config(args: argparse.Namespace) -> ImportConfig:
+    """Construct phenotype data configuration from command line arguments."""
+    config = ImportConfig()
+    config.verbose = args.verbose
+    config.instruments_dir = args.instruments
+    config.instruments_tab_separated = args.tab_separated
+
+    config.pedigree = args.pedigree
+    config.output = args.output
+
+    config.db_filename = args.pheno_db_filename
+
+    return config
+
+
+def main(argv: list[str] | None = None) -> int:
+    """Run phenotype import tool."""
+    if argv is None:
+        argv = sys.argv[1:]
+
+    try:
+        # Setup argument parser
+
+        parser = pheno_cli_parser()
+        args = parser.parse_args(argv)
+        if args.instruments is None:
+            print("missing instruments directory parameter", sys.stderr)
+            raise ValueError  # noqa: TRY301
+        if args.pedigree is None:
+            print("missing pedigree filename", sys.stderr)
+            raise ValueError  # noqa: TRY301
+        if args.pheno_name is None:
+            print("missing pheno db name", sys.stderr)
+            raise ValueError  # noqa: TRY301
+
+        if not args.browser_only:
+            import_pheno_data(args)
+
+        if not args.import_only:
+            build_browser(args)
+
+    except KeyboardInterrupt:
+        return 0
+    except ValueError as e:
+        traceback.print_exc()
+
+        program_name = "pheno_import.py"
+        indent = len(program_name) * " "
+        sys.stderr.write(program_name + ": " + repr(e) + "\n")
+        sys.stderr.write(indent + "  for help use --help")
+        return 2
+
+    return 0
+
+
+def import_pheno_data(args: Any) -> None:
+    """Import pheno data into DuckDB."""
+    os.makedirs(args.output, exist_ok=True)
+
+    args.pheno_db_filename = os.path.join(
+        args.output, f"{args.pheno_name}.db",
+    )
```

Comment:
Modifying the arguments object doesn't seem right... Separate into own variable?